<a href="https://colab.research.google.com/github/valentinamoreno4321-cpu/matrices/blob/main/Taller_para_parcial_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Parcial Parte II
## Valentina Moreno Oviedo

Instrucción 1: Genere un arreglo de NumPy con 50 números enteros aleatorios entre 1XY y 19Z,
donde XY corresponde a los dos últimos dígitos de la TI o CC y Z al último dígito de la edad.
Utilice el siguiente código:
np.random.seed(0)
datos = np.random.randint(1XY, 19Z, 50)
1. Implemente una filtración sobre el elemento datos e imprima el elemento con los valores
resultantes.
2. Escale cada uno de los valores de datos utilizando una función lambda:
𝑥􀯜 − 𝑥̅
𝑆􀯫
Para 𝑥̅ la media aritmética y 𝑆􀯫 la desviación estándar.
3. Implemente el método .describe() e interprete los resultados.
4. Convierta el elemento datos en un DataFrame usando pandas. Imprima el elemento.
5. Exporte los resultados a un archivo CSV llamado resultados.csv

In [100]:
import numpy as np
np.random.seed(0)
datos = np.random.randint(127, 191, 50)
datos
df = pd.DataFrame({
    "nota": datos
})
df


,nota
0,171
1,174
2,180
3,127
4,130
5,186
6,130
7,166
8,136
9,146


In [101]:
import pandas as pd
x = [171, 174, 180, 127, 130, 186, 130, 166, 136, 146, 148, 177, 163,
       150, 133, 151, 151, 139, 185, 128, 165, 166, 150, 173, 151, 144,
       164, 152, 140, 135, 136, 147, 178, 143, 178, 132, 189, 142, 174,
       127, 145, 162, 151, 176, 178, 156, 146, 146, 141, 166]
x = pd.DataFrame(x)
normalizar = lambda x: (x - x.mean()) / x.std()
resultado = normalizar(x)
print(resultado)


           0
0   0.929669
1   1.098495
2   1.436147
3  -1.546446
4  -1.377620
5   1.773799
6  -1.377620
7   0.648292
8  -1.039968
9  -0.477215
10 -0.364664
11  1.267321
12  0.479466
13 -0.252113
14 -1.208794
15 -0.195838
16 -0.195838
17 -0.871142
18  1.717523
19 -1.490171
20  0.592017
21  0.648292
22 -0.252113
23  1.042219
24 -0.195838
25 -0.589765
26  0.535741
27 -0.139563
28 -0.814867
29 -1.096243
30 -1.039968
31 -0.420939
32  1.323596
33 -0.646041
34  1.323596
35 -1.265069
36  1.942625
37 -0.702316
38  1.098495
39 -1.546446
40 -0.533490
41  0.423191
42 -0.195838
43  1.211045
44  1.323596
45  0.085539
46 -0.477215
47 -0.477215
48 -0.758591
49  0.648292


In [109]:
resultados = df.describe()
resultados

,nota
count,50.000000
mean,154.480000
std,17.769775
min,127.000000
25%,141.250000
50%,151.000000
75%,169.750000
max,189.000000


Podemos interpretar que

count: hace referencia a la cantidad de elementos que tenemos, es decir que tenemos 50 elementos

mean: Hace referencia a la media, es decir que nuestros datos tienen una media de 154.48

std: Representa la desviación estandar, es decir que nuestros datos tienen una desviación estandar de 17.7

min: es nuestro valor minimo es decir que nuestro valor minimo es 127

25%: se refiere que el 25% de nuestros datos son menores que  141

50%:se refiere que el 50% de nuestros datos son menores que 151

75%:se refiere que el 75% de nuestros datos son menores que 169

max: Se refiere a qu enuestro valor maximo o nuestro dato mayor es el  189

In [107]:
export = resultados.to_csv("resultados.csv")
files.download("resultados.csv")

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

Instrucción 2: Construya un DataFrame en pandas con información de su elección. El
DataFrame debe cumplir con: (1) al menos 4 columnas (por ejemplo: nombre, edad, ingreso,
ciudad), (2) al menos 5 filas (registros) y (3) incluir datos numéricos y categóricos (o variable
cualitativa)
1. Muestre las primeras filas usando método .head()
2. Seleccione una columna específica de su elección.
3. Seleccione un conjunto de filas dadas condiciones específicas sobre valores de la columna.
4. Cree una nueva columna que represente una transformación de una variable.
5. Agrupe los datos por una variable categórica usando el método .groupby(). Por ejemplo:
Agrupar por ciudad y calcular ingreso promedio
grupo = df.groupby("Ciudad")["Ingreso"].mean()
6. Exporte el DataFrame final a un archivo CSV llamado datos_procesados.csv

In [77]:
import pandas as pd
df1 = pd.DataFrame({
    "Nombre": ["Valentina", "Daniela ", "Adreiy" , "Diego" , "Paula"],
    "Edad": [21, 22, 23, 26, 20],
    "País" : ["Colombia", "Colombia", "Republica Dominicana", "Brazil" , "Costa Rica"],
    "Sexo" : ["F", "F", "F", "M", "F"]
})
df1

,Nombre,Edad,País,Sexo
0,Valentina,21,Colombia,F
1,Daniela,22,Colombia,F
2,Adreiy,23,Republica Dominicana,F
3,Diego,26,Brazil,M
4,Paula,20,Costa Rica,F


In [78]:
df1.head(1)

,Nombre,Edad,País,Sexo
0,Valentina,21,Colombia,F


In [79]:
df1["Edad"]

,Edad
0,21
1,22
2,23
3,26
4,20


In [80]:
df1.loc[df1["Edad"] > 22]

,Nombre,Edad,País,Sexo
2,Adreiy,23,Republica Dominicana,F
3,Diego,26,Brazil,M


In [81]:
df1["Edad"].apply(lambda x: x+1)

,Edad
0,22
1,23
2,24
3,27
4,21


In [82]:
grupo = df1.groupby("País")["Edad"].mean()
grupo


,Edad
País,
Brazil,26.0
Colombia,21.5
Costa Rica,20.0
Republica Dominicana,23.0


In [83]:
export = df1.to_csv("datos_procesados.csv")
files.download("datos_procesados.csv")

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

## Parte III

In [52]:
import numpy as np
import pandas as pd


datos1 = np.random.normal(loc=7.2, scale=0.4, size=56)
datos2 = np.random.gamma(shape=306, scale=0.0229, size=56)

df = pd.DataFrame({
    "numeros1": datos1
    ,"numeros2": datos2
})
df

,numeros1,numeros2
0,6.851930,6.938680
1,7.207861,6.903828
2,6.754251,7.910990
3,7.443137,7.076340
4,7.382895,7.595791
5,7.550216,7.172616
6,7.338592,6.814137
7,7.018050,6.948086
8,7.279238,7.210192
9,7.356563,7.466617


In [53]:
df.describe()

,numeros1,numeros2
count,56.000000,56.000000
mean,7.148943,7.065800
std,0.403754,0.392646
min,5.928856,6.108210
25%,6.848958,6.818874
50%,7.226147,7.012940
75%,7.385878,7.359545
max,7.984399,7.910990


In [54]:
df.median()

,0
numeros1,7.226147
numeros2,7.012940


Media: El valor de la media de los datos del grupo 1 es 7.1, el valor del grupo 2 es 7.06

Mediana: El valor de la mediana de los datos del grupo 1 es 7.22, el valor del grupo 2 es 7.01

máximo: El valor máximo de los datos del grupo 1 es 7.98, el valor del grupo 2 es 7.9

mínimo: El valor mínimo de los datos del grupo 1 es 5.92, el valor del grupo 2 es 6.10

desviación estándar: El valor de la desviación estándar de los datos del grupo 1 es 0.40, el valor del grupo 2 es  0.39

cuantil 25%: el 25% de nuestros datos son menores a 6.84 en el grupo 1 y a 6.81 en el grupo 2

cuantil 75%: el 75% de nuestros datos son menores a 7.38 en el grupo 1 y a 7.35 en el grupo 2




In [37]:
from scipy import stats

In [55]:
from scipy.stats import shapiro
shapiro(df["numeros1"])

ShapiroResult(statistic=np.float64(0.9658207095942677), pvalue=np.float64(0.11283440297205827))

como el valor p es mayor que 0.05 no se rechaza la hipotesis de normalida con una confianza del 95%

In [56]:
from scipy.stats import shapiro
shapiro(df["numeros2"])

ShapiroResult(statistic=np.float64(0.987782550353372), pvalue=np.float64(0.8406147060371743))

como el valor p es mayor que 0.05 no se rechaza la hipotesis de normalida con una confianza del 95%

In [57]:
stats.ttest_rel(df['numeros1'], df["numeros2"])

TtestResult(statistic=np.float64(1.0707058333406907), pvalue=np.float64(0.28897794222656054), df=np.int64(55))

como el valor p es mayor que 0.05 no se rechaza la hipotesis con una confianza del 95%

No se rechaza que el puntaje medio en ambas encuestas es igual con una confianza del 95%

In [58]:
stats.kendalltau(df["numeros1"], df["numeros2"])

SignificanceResult(statistic=np.float64(-0.005194805194805195), pvalue=np.float64(0.954911408492392))

como el valor p es mayor que 0.05 no se rechaza la hipotesis con una confianza del 95%

No se rechaza con una confianza del 95% que los puntajes de ambas encuestas no están
correlacionados

In [59]:
spearman = stats.spearmanr(df["numeros1"], df["numeros2"])
spearman

SignificanceResult(statistic=np.float64(-0.01394395078605605), pvalue=np.float64(0.9187579452250393))

como el valor p es mayor que 0.05 no se rechaza la hipotesis con una confianza del 95%

No se rechaza con una confianza del 95% que los puntajes de ambas encuestas no están
correlacionados

In [110]:
import os
os.makedirs("comparacion", exist_ok=True)
%%writefile comparacion/pruebat.py
%%writefile comparacion/cor.py

UsageError: Line magic function `%%writefile` not found.


In [51]:
df = pd.DataFrame({
"Estudiante": range(1, 11),
"Prueba_A": [7.1, 7.5, 6.8, 7.3, 7.0, 7.4, 6.9, 7.2, 7.6, 7.1],
"Prueba_B": [6.9, 7.2, 7.0, 7.1, 6.8, 7.3, 7.0, 7.1, 7.4, 7.0]
})